tests/integration/buildcmd/test_build_in_source.py

```diff
@@ -107,3 +115,63 @@ def test_builds_successfully_with_local_dependency(self):
 
         # check whether dependencies were installed in source dir
         self.assertEqual(os.path.isdir(os.path.join(codeuri, "node_modules")), True)
+
+
+class TestBuildCommand_BuildInSource_Nodejs(BuildIntegNodeBase):
+    is_nested_parent = False
+    template = "template.yaml"
+
+    def setUp(self):
+        super().setUp()
+
+    def tearDown(self):
+        super().tearDown()
+
+    def validate_node_modules(self, is_build_in_source_behaviour: bool):
+        # validate if node modules exist in the built artifact dir
+        built_node_modules = Path(self.default_build_dir, "Function", "node_modules")
+        self.assertEqual(built_node_modules.is_dir(), True, "node_modules not found in artifact dir")
+        self.assertEqual(built_node_modules.is_symlink(), is_build_in_source_behaviour)
+
+        # validate that node modules are suppose to exist in the source dir
+        source_node_modules = Path(self.codeuri_path, "node_modules")
+        self.assertEqual(source_node_modules.is_dir(), is_build_in_source_behaviour)
+
+    @parameterized.expand(
+        [
+            (True, True),  # build in source
+            (False, False),  # don't build in source
+            (None, False),  # use default for workflow (don't build in source)
+        ]
+    )
+    @pytest.mark.flaky(reruns=3)
+    def test_builds_successfully_without_local_dependencies(self, build_in_source, expected_built_in_source):
+        code_uri = "Node"
+        source_path = Path(self.test_data_path, code_uri)
+        self.codeuri_path = Path(self.working_dir, code_uri)
+
+        shutil.copytree(source_path, self.codeuri_path, dirs_exist_ok=True, symlinks=True)
+
+        overrides = self.get_override(
+            runtime="nodejs18.x", code_uri=self.codeuri_path, architecture="x86_64", handler="main.lambdaHandler"
+        )
+        command_list = self.get_command_list(build_in_source=build_in_source, parameter_overrides=overrides, debug=True)
+
+        run_command(command_list, self.working_dir)
+        self.validate_node_modules(expected_built_in_source)
+
+    @pytest.mark.flaky(reruns=3)
+    def test_builds_successfully_with_local_dependency(self):
+        codeuri_folder = Path("Esbuild", "NodeWithLocalDependency")
+
+        shutil.copytree(Path(self.test_data_path, "Esbuild"), self.working_dir, dirs_exist_ok=True)
+        self.codeuri_path = Path(self.test_data_path, codeuri_folder)
+        self.CODE_URI = str(codeuri_folder)
```

Comment:
Is this property used anywhere else by a parent class?

Reply:
Not anymore, this was part of a previous validation method that was never committed. Will remove to not conflict with anything in the future.